controllers/profiles/common/ensurer.go

```diff
@@ -176,6 +224,11 @@ func ensureObject(ctx context.Context, workflow *operatorapi.SonataFlow, visitor
 					return visitorErr
 				}
 			}
+			if workflow.Namespace != object.GetNamespace() {
+				// This is for Knative trigger in a different namespace
+				// Set the finalizer for trigger cleanup when the workflow is deleted
+				return setWorkflowFinalizer(ctx, c, workflow)
```

Comment:
Can you please do another check to verify if this object is a knative one?

Reply:
OK.